### PR TITLE
Avatar without email

### PR DESCRIPTION
**Pull request by jorgetwgroup**
### BEFORE - What was wrong? What was happening before this PR?

If delete or set empty email user column, gravatar fail with 500 error. Details on #4302 

### AFTER - What is happening after this PR?

Can delete or set empty email user column and not 500 error.

### How did you achieve that, in technical terms?

On helper check if gravatar is the default avatar_type, check if email column exists and is not empty, i not exists or is empty, set de default avatar_type to placehold.

### Is it a breaking change?

no

### How can we test the before & after?

Delete or set empty email column on user table.

### CODE REVIEW

**Comment by tabacitu** [comment]
Lovely ❤️ Nice and simple fix, something that we should have done AGES ago. Thanks @jorgetwgroup . 

I've assigned @pxpm to review and confirm. This is a simple PR, but it's good to get into the habit of following the process.

**Comment by pxpm** [comment]
Hey guys 👋 

Where did this `placehold` method came from ? I've never heard about it, looked at the docs and could not find it.
What this method does and what should I return from it  ?? /s

I am joking 🙃 , obviously I understand the method and what it "should be doing", but if we are adding a "feature", yes we are fixing a bug, but adding the possibility for users to build their own placeholder (even if they use emails, why not ?), I agree, but let's please add it to the docs too. 

We have other places in backpack that are quite dependant on having an email column, have a look at this recent discussion: https://github.com/Laravel-Backpack/CRUD/issues/4578#issuecomment-1216276492 
Maybe you can tackle that issue too ? If you didn't understand something ping me somewhere and we can brainstorm a bit about it. 

So, to sum up my ideas here:
- I agree with @jorgetwgroup thinking of allowing devs to "replace" the placehold in a model function /property. (I am not sure the property would work here, unless the placeholder is a static string for every model?)
- I don't like the function name, I think `placeholder` or `modelPlaceholder` or `userPlaceholder/userPlacehold` (technically they are always users with or without email) are better descriptors (I like more the last ones)
- I don't agree that we only allow the dev to override if there is no mail.
    - I think that if `userPlaceholder()` is defined on the model we should use it, no matter if it has email or not. 
- I think this needs to be documented. 


Let me know what you think about my suggestions @jorgetwgroup 

Cheers

**Comment by tabacitu** [comment]
OOoohhh I did not notice we're calling a new `placehold()` method. My 2 cents - fix this quick, without adding a new method, if that's possible. Then you can talk about method names, how to add customization etc to death... in a new issue in the IDEAS repo. But keep in mind that becomes a COULD-DO.

**Comment by pxpm** [comment]
I agree with @tabacitu .

If you are on the same page as me and @tabacitu please let's just fix the if condition to don't cause the bug and open a new PR with the else code and the feature introduction.
If some default value is mandatory let it be an empty string so that we don't change other files ahead in the application lifecycle. 
Cheers

**Comment by jorgetwgroup** [comment]
Hello @pxpm @tabacitu i didnt create any new function, thats i copy is the behaviour if you set "placehold" in avatar_type on config/backpack/base.php, if you go to helper and check default will be this:

`default:
                return method_exists($user, config('backpack.base.avatar_type')) ? $user->{config('backpack.base.avatar_type')}() : $user->{config('backpack.base.avatar_type')};
                break;`

i do the same, just set "placehold" as default in case there is no email and config was set on gravatar, but user can set anything they want it in base.php and everything will work as they want it.

I have experience with this functions because in our platform we disable gravatar many times, because for some reason, sometimes take many seconds to load image, then we replace with our own local avatar creator.

Please let me know exactly what i need to change to make this comfortable for all us, thanks.

Cheers.

**Comment by ziming** [comment]
an outsider opinion here. hope it's not too irrelevant

i think the https://ui-avatars.com/ used by laravel jetstream could be a good placeholder choice. it is more privacy friendly than gravatar as well. the documentation also provide an example of its use as a gravatar fallback

i know this PR is a bug fix and not about a new feature so that's all i'm going to say.

**Comment by jorgetwgroup** [comment]
hi @ziming thanks for suggestion, we will check this as different idea, this is focus to fix bug on no email.

@pxpm i fix method directly to our "$user->placehold", user is free to choose wherever method they want it config this on base.php

`'avatar_type' => 'OWN_METHOD'`

this fix, just check if method is gravatar and not email column is present, in this scenario, use placehold method as alternative and take "authentication_column" as input to create placehold for user.

About #4578 i check discussion but they are almost speaking about not use database, thats is major update.

Let me know any feedback about it.

Cheers.

**Comment by tabacitu** [comment]
@jorgetwgroup are you sure? Maybe you didn't push the changes? I see this is still calling `$user->placehold`, which will error in 99.99% projects. So it doesn't fall back to the `avatar_type`, it falls back to the hardcoded `$user->placehold`.

**Comment by pxpm** [comment]
Hey guys. 
I think we are having a miscomunication here, our final goal is the same 👍 

@jorgetwgroup I think we should remove the `else` completely. Why?
- If you don't return nothing from the helper, same as returning `->placehold` (would be **null** if user has `gravatar` in config) we will endup with an `` tag with an error in the source. 
image
Sometime ago, we implemented a feature that when this image have an error we hide it and display a css based solution that show the user initials (like the old placehold.it service we were using that failed from time to time). 

https://github.com/Laravel-Backpack/CRUD/blob/d1966268d6e292e4ed7e68c777c7a3450dbcf409/src/resources/views/base/inc/menu_user_dropdown.blade.php#L3

Let me know if I made myself clear here. 

Cheers

**Comment by jorgetwgroup** [comment]
You right, i hit my head against a rock, now else was deleted.

Cheers

**Comment by pxpm** [comment]
Thank you @jorgetwgroup 👍
